src/data/list/palindrome.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2020 Google LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Wong
+-/
+
+import data.list.basic
+open list
+
+/-!
+# Palindromes
+
+This module defines *palindromes*, lists which are equal to their reverse.
+
+The main result is the `palindrome` inductive type, and its associated `palindrome.rec_on` induction
+principle. Also provided are conversions to and from other equivalent definitions.
+
+## References
+
+* [Pierre Castéran, *On palindromes*][casteran]
+
+[casteran]: https://www.labri.fr/perso/casteran/CoqArt/inductive-prop-chap/palindrome.html
+
+## Tags
+
+palindrome, reverse, induction
+-/
+
+variables {α : Type*}
+
+/--
+`palindrome l` asserts that `l` is a palindrome. This is defined inductively:
```

Comment:
Should this be `is_palindrome` or `palindromic`? What's the mathlib convention on nouns vs adjectives for `Prop`s?

Reply:
Good point. I couldn't find anything on the [naming conventions](https://leanprover-community.github.io/contribute/naming.html) page about this.

My reasoning for `palindrome` is that (a) the `is_` prefix doesn't add anything, and (b) I'm more likely to search for `palindrome` than `palindromic`.

I guess the closest analog would be [`perm`](https://leanprover-community.github.io/mathlib_docs/data/list/perm.html) which is also an inductive `Prop` on lists. Using `palindrome` would be closest to that in spirit (ignoring silly options like `pali`).

Reply:
Found a few more in [`data.list.defs`](https://leanprover-community.github.io/mathlib_docs/data/list/defs.html). There's `chain` (noun/verb), `pairwise` (adjective/adverb), `​forall₂` (???) so it's all over the place.

Reply:
I'm going to resolve this, as I think the name is in line with other similar definitions in `data.list`.

Reply:
Those names are not really the same, as they're functions which take a predicate - `pairwise is_even` and `chain is_even` can be thought of as the names of the predicates over lists.

I think we should leave this unresolved to get another pair of eyes - even if no change to the PR ends up being needed.

Reply:
`is_infix`, `is_suffix`, and `is_prefix` are examples of the naming going the other way, with `nodup` being abbreviated to the point that it's hard to classify.

Reply:
It does seem like we're inconsistent on this. I recommend opening a Zulip thread for some discussion.

Reply:
Zulip thread at https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Noun.2Fadjective.2Fverb.20naming.20conventions.20for.20defs.20in.20mathlib/near/206742747